### ISSUE TITLE
Node panics when deploying a duplicate contract identifier

### ISSUE THREAD

**Issue by hstove** [blockstack/stacks-blockchain#1411]
To repro:

- Publish a contract
- Try and publish the contract again, with the same exact contract identifier. I think it doesn't matter if you change the source code or not.

Backtrace:

```
WARN [1586545211.758] [src/chainstate/stacks/db/transactions.rs:451] Duplicate contract 'ST22T6ZS7HVWEMZHHFK77H4GTNDTWNPQAX8WZAKHJ.status'
ERROR [1586545211.758] [src/testnet/helium/tenure.rs:66] Failed mining transaction - Duplicate contract 'ST22T6ZS7HVWEMZHHFK77H4GTNDTWNPQAX8WZAKHJ.status'
Burnchain block #6 (4391a5c79ffdc79883036503ca551673c09deec28df432a8d88debc7fa2ec91e) was produced with sortition #a68d72666f313c39a1a2acbec568bfb16238dd69bb88d0495b5b0f9a78e4126b
WARN [1586545211.854] [src/chainstate/stacks/db/blocks.rs:2692] Block 16fbcd9cb28398f05e94107d2a1b2fb49216d327344091b27aa06aa4b40ef35d state root mismatch: expected 35b48b8437deab7c70b30c51218fed5c1ff3887fec09aef1deec1d56db234aaa, got 3de07d140c1d8bb39775b9854922586468cf8ac68dc009e3bca16fb8584e963a
WARN [1586545211.866] [src/chainstate/stacks/db/blocks.rs:2962] Encountered invalid block: Block 16fbcd9cb28398f05e94107d2a1b2fb49216d327344091b27aa06aa4b40ef35d state root mismatch: expected 35b48b8437deab7c70b30c51218fed5c1ff3887fec09aef1deec1d56db234aaa, got 3de07d140c1d8bb39775b9854922586468cf8ac68dc009e3bca16fb8584e963a
thread 'main' panicked at 'index out of bounds: the len is 0 but the index is 0', /rustc/b8cedc00407a4c56a3bda1ed605c6fc166655447/src/libcore/slice/mod.rs:2791:10
stack backtrace:
   0: std::sys_common::at_exit_imp::push
   1: core::fmt::ArgumentV1::show_usize
   2: std::io::Write::write_fmt
   3: std::panicking::default_hook::{{closure}}
   4: std::panicking::default_hook
   5: <std::panicking::begin_panic::PanicPayload<A> as core::panic::BoxMeUp>::get
   6: std::panicking::try::do_call
   7: core::panicking::panic_fmt
             at src/libcore/panicking.rs:85
   8: core::panicking::panic_bounds_check
             at src/libcore/panicking.rs:63
   9: <usize as core::slice::SliceIndex<[T]>>::index
             at /rustc/b8cedc00407a4c56a3bda1ed605c6fc166655447/src/libcore/slice/mod.rs:2791
  10: core::slice::<impl core::ops::index::Index<I> for [T]>::index
             at /rustc/b8cedc00407a4c56a3bda1ed605c6fc166655447/src/libcore/slice/mod.rs:2656
  11: <alloc::vec::Vec<T> as core::ops::index::Index<I>>::index
             at /rustc/b8cedc00407a4c56a3bda1ed605c6fc166655447/src/liballoc/vec.rs:1883
  12: blockstack_lib::testnet::helium::node::Node::process_tenure
             at src/testnet/helium/node.rs:425
  13: blockstack_lib::testnet::helium::run_loop::RunLoop::start
             at src/testnet/helium/run_loop.rs:193
  14: blockstack_core::main
             at src/main.rs:206
  15: std::rt::lang_start::{{closure}}
             at /rustc/b8cedc00407a4c56a3bda1ed605c6fc166655447/src/libstd/rt.rs:67
  16: std::panicking::try::do_call
  17: panic_unwind::dwarf::eh::read_encoded_pointer
  18: <std::panicking::begin_panic::PanicPayload<A> as core::panic::BoxMeUp>::get
  19: std::rt::lang_start
             at /rustc/b8cedc00407a4c56a3bda1ed605c6fc166655447/src/libstd/rt.rs:67
  20: blockstack_core::main
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

**Comment by diwakergupta** [comment]
Should be resolved by #1376

**Comment by diwakergupta** [comment]
Fixed by #1415
